sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/TestCases/TypeNamesShouldNotMatchNamespaces.cs

```diff
@@ -0,0 +1,21 @@
+using System;
+
+namespace Tests.Diagnostics
+{
```

Comment:
This is going to be boring but I would add a test case for every name we handle (avoid regression).